keps/sig-storage/1698-generic-ephemeral-volumes/README.md

```diff
@@ -564,8 +569,24 @@ Will be added before the transition to beta.
 
 - Kubernetes 1.19: alpha (tentative)
 
+## Drawbacks
+
+Allowing users to create PVCs indirectly through pod creation is a new
+capability which must be considered in a cluster's security policy. As
+explained in [risks and mitigations above](#risks-and-mitigations),
+the feature has to be configurable on a per-user basis.
+
+Making a pod the owner of PVC objects introduces a new dependency
+which may cause problems in code that expects that pods can always be
+deleted immediately. One example was the PVC protection controller
+which waited for pod deletion before allowing the PVC to be deleted:
+it had to be extended to allow PVC deletion also when the pod had
+terminated, but not deleted yet.
```

Comment:
How does the controller deal with a race condition of deleting a pod, the PV still existing, and a new pod with a different PV with the same name being created?  Which controller manages that?

Reply:
There is going to be a new controller in kube-controller-manager which handles PVC creation. These PVCs are tied to the pod through the owner reference. A PVC originally created for the already deleted pod is recognized as not belonging to the new pod because it has the wrong owner reference. It will have to be deleted before the new pod can start up, because the PVC for the new pod cannot be created. The old PVC will be deleted by the garbage collector.

That covers the PVC lifecycle. PV creation and deletion is tied to those PVCs through the usual provisioning and de-provisioning process.

PVC deletion is not blocked by PV deletion, so in practice, the duration during which an old PVC can block startup of a new pod should be limited.

Reply:
> PVC deletion is not blocked by PV deletion, so in practice, the duration during which an old PVC can block startup of a new pod should be limited.

It's worth adding that this is only an issue for pods with stable names. App controllers use `GenerateName` for the pods automatically end up also with a new name for the PVC.